src/App.tsx

```diff
@@ -15,48 +17,143 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET: {
+      visibleGoods.sort((a, b) => a.localeCompare(b));
+      break;
+    }
+
+    case SortType.LENGTH: {
+      visibleGoods.sort((a, b) => a.length - b.length);
+      break;
+    }
+
+    case SortType.NONE:
+    default: {
+      break;
+    }
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const sortAlphabetically = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const reverse = () => {
+    setReversed((prevReversed) => !prevReversed);
+  };
+
+  const reset = () => {
+    setSortType(SortType.NONE);
+    setReversed(false);
+  };
+
+  const reorderedGoods = getReorderedGoods(goodsFromServer, {
+    sortType,
+    isReversed,
+  });
+
+  const isOriginalOrder
+    = goodsFromServer.every((item, index) => item === reorderedGoods[index]);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button',
+            'is-info',
+            {
+              'is-light': sortType !== SortType.ALPHABET,
+            },
+          )}
+          onClick={() => sortAlphabetically()}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames(
+            'button',
+            'is-success',
+            {
+              'is-light': sortType !== SortType.LENGTH,
+            },
+          )}
+          onClick={() => sortByLength()}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames(
+            'button',
+            'is-warning',
+            {
+              'is-light': !isReversed,
+            },
+          )}
+          onClick={() => reverse()}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {!isOriginalOrder && (
+          <>
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={() => reset()}
```

Comment:
```suggestion
              onClick={reset}
```
no need to make arrow anonymous function here. just pass reference